src/client/Microsoft.Identity.Client/IntuneAppProtectionPolicyRequiredException.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// This exception is thrown when Intune requires App protection policy.
+    /// Properties in it can be used by app to obtain the required enrollmentID from MAM SDK
+    /// </summary>
+    public class IntuneAppProtectionPolicyRequiredException : MsalServiceException
+    {
+        /// <summary>
+        /// UPN of the user
+        /// </summary>
+        public string Upn { get; internal set; }
```

Comment:
Please make setters public. All our public API should be testable, and since concrete types can't be mocked, the simplest is to make sure properties have public setters.